### PR TITLE
Don't log warnings for 404 errors by default

### PR DESCRIPTION
**Pull request by markus-moser**
This is a alternative approach for #5 to ignore 404 errors by default. This PR is done with a custom monolog handler which filters logs which should not be displayed.

### CODE REVIEW

File: src/OpenSearchClientFactory.php [hunk 1]
```diff
@@ -26,6 +28,11 @@ public static function createOpenSearchClient(LoggerInterface $logger, array $co
     {
         $clientBuilder = new ClientBuilder();
         $clientBuilder->setHosts($config['hosts']);
+
+        if(!$config['log_404_errors'] && $logger instanceof Logger) {
```
Comment (martineiber) [inline src/OpenSearchClientFactory.php]:
```suggestion
        if (!$config['log_404_errors'] && $logger instanceof Logger) {
```

**Review by martineiber — CHANGES_REQUESTED** [review]

**Review by martineiber — APPROVED** [review]
Rest LGTM :)

**Comment by martineiber** [comment]
We should update the doc as well?

**Comment by markus-moser** [comment]
> We should update the doc as well?

Thanks for updating the docs!

File: src/OpenSearchClientFactory.php (hunk 1 above)
Comment (markus-moser) [reply src/OpenSearchClientFactory.php]: Thanks, fixed now!

**Pull request merged**